pkgs/dart_tooling_mcp_server/lib/src/mixins/dtd.dart

```diff
@@ -24,10 +24,50 @@ base mixin DartToolingDaemonSupport on ToolsSupport {
   /// ready to be invoked.
   bool _getDebugSessionsReady = false;
 
+  /// A Map of [VmService] objects by their associated VM Service URI
+  /// (represented as a String).
+  ///
+  /// [VmService] objects are automatically removed from the Map when the
+  /// [VmService] shuts down.
+  final _activeVmServices = <String, VmService>{};
```

Comment:
@jakemac53 it doesn't look like there is a good way to test the logic inside of the tool mixins. All that is available from the test harness is calling the registered tools. Do you see any issues with running some tests in process so that we can test the implementation logic inside the server tools? I see [here](https://github.com/dart-lang/ai/blob/main/pkgs/dart_tooling_mcp_server/test/test_harness.dart#L331) that we create the DartToolingMCPServer in process, but this is currently limited to `debugMode` only.

Reply:
I think it's fine if we want to run some tests this way explicitly so we can inspect the server state - we just don't want to run all tests that way IMO.

I would probably rename the parameter though in that case.

Reply:
Compiling the MCP server right now is also likely most of the test time, since it's happening for each test.

We might want to consider moving more tests to the in process mode and just having a single end to end test which does some basic interactions (just list the tools for instance). Really we just need to validate that the server actually works as a separate process.